system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -775,6 +789,352 @@ extern ICompiledXpath* compileXpath(const char * xpath)
+//        parserCtx->myDoc = doc;
+        parserCtx->node = sect;
+        xmlParseDocument(parserCtx);
+        int wellFormed = parserCtx->wellFormed;
+        xmlFreeDoc(parserCtx->myDoc); //dummy document
+        xmlFreeParserCtxt(parserCtx);
+        if (!wellFormed)
+           throw MakeStringException(-1, "CEsdlScriptContext:setContent xml string: Unable to parse %s XML content", section);
+    }
+    virtual void setContent(const char *section, IPropertyTree *tree) override
+    {
+        xmlNodePtr sect = replaceSection(section);
+        if (tree==nullptr) //means delete content
+            return;
+        addChildFromPtree(sect, *tree, tree->queryName());
+    }
+
+    virtual void setAttribute(const char *section, const char *name, const char *value) override
+    {
+        xmlNodePtr sect = ensureSection(section);
+        xmlNewProp(sect, (const xmlChar *)name, (const xmlChar *)value);
+    }
+    virtual const char *queryAttribute(const char *section, const char *name) override
+    {
+        xmlNodePtr sect = getSection(section);
+        if (!sect)
+            return nullptr;
+        return queryAttrValue(sect, name);
+    }
+    virtual const char *getAttribute(const char *section, const char *name, StringBuffer &s) override
+    {
+        xmlNodePtr sect = getSection(section);
+        if (!sect)
+            return nullptr;
+        xmlChar *val = xmlGetProp(sect, (const xmlChar *)name);
+        if (!val)
+            return nullptr;
+        s.append((const char *)val);
+        xmlFree(val);
+        return s;
+    }
+
+    virtual void toXML(StringBuffer &xml, const char *section, bool includeParentNode=false) override
+    {
+        xmlNodePtr sect = root;
+        if (!isEmptyString(section))
+        {
+            sect = getSectionNode(section, includeParentNode ? nullptr : "*[1]");
+            if (!sect)
+                return;
+        }
+
+        xmlOutputBufferPtr xmlOut = xmlAllocOutputBuffer(nullptr);
+        xmlNodeDumpOutput(xmlOut, sect->doc, sect, 0, 1, nullptr);
+        xmlOutputBufferFlush(xmlOut);
+        xmlBufPtr buf = (xmlOut->conv != nullptr) ? xmlOut->conv : xmlOut->buffer;
+        if (xmlBufUse(buf))
+            xml.append(xmlBufUse(buf), (const char *)xmlBufContent(buf));
+        xmlOutputBufferClose(xmlOut);
+
+    }
+    virtual void toXML(StringBuffer &xml) override
+    {
+        toXML(xml, nullptr, true);
+    }
+    IXpathContext* createXpathContext(const char *section, bool strictParameterDeclaration) override
+    {
+        xmlNodePtr sect = getSectionNode(section);
+        if (!sect)
+            throw MakeStringException(-1, "CEsdlScriptContext:createXpathContext: section not found %s", section);
+        CLibXpathContext *xpathContext = new CLibXpathContext(doc, sect, strictParameterDeclaration);
+        StringBuffer val;
+        xpathContext->addVariable("method", getAttribute("esdl", "method", val));
```

Comment:
Why did you spit  up the initialization of the xpathContext between this createXpathContext() function and the processServiceAndMethodTransforms() method? I found it nice to see all the context setup in that one function- made it easy to tell which values were treated like parameters and which like implicit variables.

Reply:
I thought these values could be understood by the script context, but wasn't sure about some others.  We can consider reorganizing further in the future.